### PR TITLE
Use `Self` in docs when possible

### LINKED ISSUE #76542: More Self usage in PartialOrd trait docs example
In this docs page:
https://doc.rust-lang.org/nightly/std/cmp/trait.PartialOrd.html

There's this code:

```rust
use std::cmp::Ordering;

#[derive(Eq)]
struct Person {
    id: u32,
    name: String,
    height: u32,
}

impl PartialOrd for Person {
    fn partial_cmp(&self, other: &Person) -> Option<Ordering> {
        Some(self.cmp(other))
    }
}

impl Ord for Person {
    fn cmp(&self, other: &Person) -> Ordering {
        self.height.cmp(&other.height)
    }
}

impl PartialEq for Person {
    fn eq(&self, other: &Person) -> bool {
        self.height == other.height
    }
}
```

But I think docs must show the best and up to date idioms, so I think it should be replaced with (some `Person` replaced by more DRY `Self`):

```rust
use std::cmp::Ordering;

#[derive(Eq)]
struct Person {
    id: u32,
    name: String,
    height: u32,
}

impl PartialOrd for Person {
    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
        Some(self.cmp(other))
    }
}

impl Ord for Person {
    fn cmp(&self, other: &Self) -> Ordering {
        self.height.cmp(&other.height)
    }
}

impl PartialEq for Person {
    fn eq(&self, other: &Self) -> bool {
        self.height == other.height
    }
}
```

### PR DESCRIPTION
**Pull request by poliorcetics**
Fixes #76542.

I used `rg '\s*//[!/]\s+fn [\w_]+\(&?self, ' .` in `library/` to find instances, I found some with that and some by manually checking.

@rustbot modify labels: C-enhancement T-doc

### CODE REVIEW

File: library/core/src/ops/bit.rs [hunk 1]
```diff
@@ -640,11 +643,11 @@ shr_impl_all! { u8 u16 u32 u64 u128 usize i8 i16 i32 i64 i128 isize }
...
-///         *self = BooleanVector(self.0
-///                                   .iter()
-///                                   .zip(rhs.0.iter())
-///                                   .map(|(x, y)| *x && *y)
-///                                   .collect());
+///         *self = Self(self.0
+///                          .iter()
+///                          .zip(rhs.0.iter())
+///                          .map(|(x, y)| *x && *y)
+///                          .collect());
```
Comment (jyn514) [inline library/core/src/ops/bit.rs:650]: rustfmt continues to disappoint me in every way 😢

**Review by jyn514 — APPROVED** [review]
r=me with CI passing

**Comment by jyn514** [comment]
Looks like it's already passing.

@bors r+ rollup

**Comment by pickfire** [comment]
Would this be good? In the docs it is hard to identify what `Self` is, if there is multiple implementations `Self` could be a few things and it could confuse the readers. Maybe we should hold this off until we have a clearer way to identify `Self`?

**Comment by jyn514** [comment]
@pickfire this is in doc-tests, so the types will still be shown in the documentation itself.

**Comment by poliorcetics** [comment]
In doc tests blocks and functions are usually very short and contained so I don't think this is going to be a problem.

**Comment by pickfire** [comment]
> @pickfire this is in doc-tests, so the types will still be shown in the documentation itself.

Yes, I know doctests have less of this issue but it encourages user to write it this way, which in turns make the ecosystems' docs harder to read.

**Pull request merged**